src/js_realm_object.hpp

```diff
@@ -162,6 +164,29 @@ typename T::Object RealmObjectClass<T>::create_instance(ContextType ctx, realm::
     }
 }
 
+template <typename T>
+void RealmObjectClass<T>::constructor(ContextType ctx, ObjectType this_object, Arguments& args)
+{
+    // Parse aguments
+    args.validate_between(1, 2);
+    auto constructor = Object::validated_get_object(ctx, this_object, "constructor");
+    auto realm = Value::validated_to_object(ctx, args[0], "realm");
+    auto values = Value::is_undefined(ctx, args[1]) ? Object::create_empty(ctx) : args[1];
+
+    // Create an object
+    std::vector<ValueType> create_args{constructor, values};
+    Arguments create_arguments{ctx, create_args.size(), create_args.data()};
+    ReturnValue result{ctx};
+    RealmClass<T>::create(ctx, realm, create_arguments, result);
+    ObjectType realm_object_object = Value::validated_to_object(ctx, result);
```

Comment:
I love this variable name :D So this is a c++ Realm object transformed to a JS object?

Reply:
Yep. This is a JS object wrapping a C++ Realm object.
To be honest, this name is be cause I have another `realm_object` below 🙂 

I could rename this to `js_realm_object`?

Reply:
That would be way less funny 😆  But more clear

Reply:
I think `js_realm_object` is clearer also

Reply:
I renamed to `js_realm_object`.

Reply:
Since this is thrown away and only used to read its "internal", perhaps I should call it `temporary_realm_object` or `throwaway_realm_object` instead?

Reply:
`tmp_realm_object`